packages/providers/vercel/src/index.ts

```diff
@@ -13,11 +13,118 @@
 import { BaseAgenticProvider, Tool as ComposioTool, ExecuteToolFn } from '@composio/core';
 import type { Tool as VercelTool } from 'ai';
 import { jsonSchema, tool } from 'ai';
+import { MCPToolkitConfig, MCPAuthOptions, MCPGetServerParams } from '@composio/core/src/types/mcp.types';
+import { McpServerCreateResponse, McpUrlResponse, McpProvider } from '@composio/core/src/provider/BaseProvider';
 
 type VercelToolCollection = Record<string, VercelTool>;
-export class VercelProvider extends BaseAgenticProvider<VercelToolCollection, VercelTool> {
+
+interface VercelMCPResponse {
+  transport: "sse";
+  url: URL;
+}
+
+export class VercelMcpProvider extends McpProvider<VercelMCPResponse[]> {
   readonly name = 'vercel';
 
+  protected transformGetResponse(
```

Comment:
This is repeated code across all the providers now, we should be ideally handling this in the core itself. These are core logic overflowing into providers, ideally the provider should have as little context and logic as possible.

Reply:
This is repeated at each provider because, every provider has it's own return type for MCP Create response,